inception-search-mtas/src/test/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndexTest.java

```diff
@@ -311,8 +313,38 @@ public void testAnnotationQuery() throws Exception
 
         uploadDocument(project, fileContent, sourceDocument);
 
+        // Wait for the asynchronous indexing task to finish. We need a sleep before the while 
+        // because otherwise there would not be time even for the index becoming invalid 
+        // before becoming valid again.
+        
+        Thread.sleep(WAIT_TIME);
+        
+        int numWaits = 0;
+        
+        while (!searchService.isIndexValid(project) && numWaits < NUM_WAITS) {
+            log.debug("Waiting for the annotation to be indexed...");
```

Comment:
You could use awaitility (https://github.com/awaitility/awaitility), as the currently implemented variant really slows down unit test execution time (10sec min, up to 30sec here?).

Reply:
All of this is work in progress, the test is still not working well, I'm trying it with different wait times.

Reply:
In pure test code, I'd not use the logging framework at all... just log via System.out.

Reply:
@betoboullosa how about generating a Spring event when the indexing of a document is complete, then you can wait for this event.

Reply:
@reckart You mean always generating this event just for passing the test?

Reply:
The event could also be used for other purposes as well, e.g. to update a progress bar in the UI or to display a notification in the UI that new search results are available and the user may refresh the search. But for the moment it would only be used by the test, yes.